tests/testSBMLSuite.py

```diff
@@ -73,93 +72,68 @@ def test_sbml_testsuite_case(test_number, result_path):
     # hasOnlySubstanceUnits="true", but then request results as concentrations.
-    requested_concentrations = [s
-            for s in settings['concentration'].replace(
-                ' ', '').replace('\n', '').split(',') if s]
+    requested_concentrations = [
+        s for s in
+        settings['concentration'].replace(' ', '').replace('\n', '').split(',')
+        if s
+    ]
     # The rate rules condition here may be unnecessary/better implemented
     # elsewhere.
-    concentration_species = [species for species in requested_concentrations
-            if wrapper.species_has_only_substance_units[
-                wrapper.species_index[species]]
-            and species in [str(s) for s in wrapper.species_rate_rules]]
-    amounts_to_concentrations(concentration_species, wrapper, model,
-            simulated_x, rdata['y'], requested_concentrations)
-
-    concentrations_to_amounts(amount_species, wrapper, model,
-            simulated_x, rdata['y'], requested_concentrations)
-
-    # Add observables to the verification. This includes compartments with
-    # assignment rules, as they are implemented as observables.
-    # Currently only used for SBML test suite case 01223.
-    observables = {
-        str(o_id): (o_index, variables_species.index(str(o_id)))
-        for o_index, o_id in enumerate(
-            wrapper.symbols['observable']['identifier']
-        )
-        if str(o_id) in variables_species
-        and (str(o_id) not in wrapper.species_index)
-    }
-    for o_id, indices in observables.items():
-        simulated_x = np.hstack((simulated_x, np.array([rdata['y'][:,
-                                                        indices[0]]]).T))
-        expected_x = np.hstack((expected_x, np.array([results[1:,
-                                                      1+indices[1]]]).T))
-        x_ids.append(o_id)
-
-    assert np.isclose(simulated_x, expected_x, atol, rtol).all()
-
-    # TODO: verify compartment volumes and parameters
-    # Currently, compartments with assignment and rate rules are verified.
-    # Compartments with assignment rules are exported as observables, and
-    # compartments with rate rules are exported as species.
-
-    return simulated_x, x_ids
+    concentration_species = [
+        species for species in requested_concentrations
+        if wrapper.species_has_only_substance_units[
+               wrapper.species_index[species]
+        ] and species in [str(s) for s in wrapper.species_rate_rules]
+    ]
+    amounts_to_concentrations(concentration_species, wrapper,
+                              simulated, requested_concentrations)
+
+    concentrations_to_amounts(amount_species, wrapper, simulated,
+                              requested_concentrations)
+    for variable in variables:
+        # some values in expected will be int64, so we cast everything as float
+        # to avoid errors from downcasting float64 to int64
```

Comment:
Not converted automatically? :flushed:

Reply:
simulated has dtype `object` for which np.isclose doesnt work and expected is mixed `np.int64`, `np.float64` so you have to explicitely cast. Not sure what happens when comparing `int64` vs `float64`, but that isn't easy to implement, but comment may indeed be a bit misleading.